clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/view/MaterializedViewWriter.java

```diff
@@ -67,34 +67,35 @@ public CompletableFuture<Void> processRecord(
       int oldValueSchemaId,
       GenericRecord replicationMetadataRecord,
       Lazy<GenericRecord> valueProvider) {
-    return processRecord(newValue, key, newValueSchemaId, false, valueProvider);
+    return processRecord(newValue, key, newValueSchemaId, null, valueProvider);
   }
 
   /**
-   * Before we have proper chunking support for view writers we assume that even when chunking is enabled the actual
-   * k/v will not be chunked. This way we don't need to worry about how to ensure all the chunks are forwarded to the
-   * same view partition during NR's pass-through mode. Proper chunking support can leverage message footer populated
-   * by the CompositeVeniceWriter in VPJ (write to views first and then VT) to figure out which view partition to
-   * forward the chunks to. Another alternative is trigger the view writer write upon receiving the manifest, and we
-   * will assemble and re-chunk.
+   * During NR pass-through viewPartitionSet is going to be provided. This way we can forward record or chunks of a
+   * record to the appropriate view partition without the need to assemble or repartition.
    */
   @Override
   public CompletableFuture<Void> processRecord(
       ByteBuffer newValue,
       byte[] key,
       int newValueSchemaId,
-      boolean isChunkedKey,
+      Set<Integer> viewPartitionSet,
       Lazy<GenericRecord> newValueProvider) {
-    byte[] viewTopicKey = key;
-    if (isChunkedKey) {
-      viewTopicKey = keyWithChunkingSuffixSerializer.getKeyFromChunkedKey(key);
-    }
     byte[] newValueBytes = newValue == null ? null : ByteUtils.extractByteArray(newValue);
+    if (viewPartitionSet != null) {
```

Comment:
So..... a pending problem we have in incremental push is that we don't have support for sending DELETE events from inc push.  We used to, but there were problems in adding the DELETE type to the union.

The way I had kind of figured we'd handle this is we could just send null values....  This work isn't planned, implemented, or even design reviewed.  But I'm trying to think through if this puts us in a corner or not.

Reply:
I thought we moved incremental push to writing to RT exclusively a long time already. DELETE via incremental push will not be in this branch and should be supported by L/F or A/A SIT's RT consumption code path.